alibi/explainers/anchor_tabular.py

```diff
@@ -958,30 +979,64 @@ def add_names_to_exp(self, explanation: dict) -> None:
                 handled.add(feat_id)
             explanation['names'].append(fname)
 
+    @property
+    def predictor(self) -> Callable:
+        return self._ohe_predictor if self.ohe else self._predictor
+
+    @predictor.setter
+    def predictor(self, predictor: Callable) -> None:
+        # if input is one-hot encoded
+        if self.ohe:
+            # this predictor expects ordinal/labels encoded categorical variables
+            ord_predictor = lambda x: predictor(ord_to_ohe(x, self.cat_vars_ord)[0])  # noqa: E731
+            self._predictor = self._transform_predictor(ord_predictor) if predictor else None
+
+            # this predictor expects one-hot encoded categorical variable
+            self._ohe_predictor = self._transform_ohe_predictor(predictor) if predictor else None
+
+        else:
+            # set the predictor
+            self._predictor = self._transform_predictor(predictor) if predictor else None
+
     def _transform_predictor(self, predictor: Callable) -> Callable:
-        # check if predictor returns predicted class or prediction probabilities for each class
-        # if needed adjust predictor so it returns the predicted class
-        if np.argmax(predictor(np.zeros([1, len(self.feature_names)])).shape) == 0:
+        # define data instance full of zeros
+        zeros = np.zeros([1, len(self.feature_names)])
+
+        try:
+            # check if predictor returns predicted class or prediction probabilities for each class
+            # if needed adjust predictor so it returns the predicted class
+            prediction = predictor(zeros)
+        except Exception:
+            ncols = ord_to_ohe(zeros, self.cat_vars_ord)[0].shape[1] if self.ohe else zeros.shape[1]
+            raise ValueError("The classifier is expecting a different number of features. "
+                             f"Calling a classifier using {ncols} failed, check the values"
+                             "of the `feature_names` and `categorical_names`.")
```

Comment:
Please expand slightly to `using {ncols} features` and add a space after `check the values " just like in the first line.